File: 0997-find-the-town-judge/0997-find-the-town-judge.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust)
    {
        vector <int> degree(n+1, 0);

        vector <vector <int>> graph(n+1);

        for (auto &t : trust)
        {
            graph[t[0]].push_back(t[1]);

            degree[t[1]] ++;
        }

        for (int node = 1; node <= n; node ++)
        {
            if (graph[node].empty() and degree[node] == n-1)
            {
                return node;
            }
        }

        return -1;
    }
};

File: 0997-find-the-town-judge/0997-find-the-town-judge.java
class Solution {
    public int findJudge(int n, int[][] trust) {

        List<List<Integer>> adj = new ArrayList<>();

        for (int i = 0; i <= n; i++) {
            adj.add(new ArrayList<>());
        }

        int[] inDegree = new int[n + 1];

        for (int[] edge : trust) {
            int person = edge[0];
            int trustedPerson = edge[1];

            adj.get(person).add(trustedPerson);

            inDegree[trustedPerson]++;
        }

        for (int person = 1; person <= n; person++) {

            boolean trustsNobody = adj.get(person).isEmpty();

            boolean everyoneTrusts = inDegree[person] == n - 1;

            if (trustsNobody && everyoneTrusts) {
                return person;
            }
        }

        return -1;
    }
}